### PR TITLE
Added implementation for the Global Exception Handler feature

### PR DESCRIPTION
**Pull request by MinoDentori**
1. Added hibernate-validator dependency.
2. Added validator annotations to the CreateRequestBookDto class
3. Added @Validate annotation to the save, and update methods in the Controller class.
4. Created CustomGlobalHandlerException class to handle exceptions thrown by the validator.

### CODE REVIEW

**Review by dima-semeniuk — APPROVED** [review]
Looks good, but think about adding to CustomGlobalExcHandler method to handle EntityNotFoundException

File: src/main/java/mate/academy/onlinebookstore/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 @Data
 public class CreateBookRequestDto {
+    @NotNull
     private String title;
+    @NotNull
     private String author;
+    @NotNull
     private String isbn;
+    @NotNull
+    @Min(0)
```
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstore/dto/CreateBookRequestDto.java]:
```suggestion
    @Positive
```

File: src/main/java/mate/academy/onlinebookstore/dto/CreateBookRequestDto.java
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstore/dto/CreateBookRequestDto.java:1]: please provide your custom informative messages :)

File: src/main/java/mate/academy/onlinebookstore/exception/CustomGlobalExceptionHandler.java
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstore/exception/CustomGlobalExceptionHandler.java:1]: you should also write `@ExceptionHandler` for your `EntityNotFoundException`

File: src/main/java/mate/academy/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex,
+            HttpHeaders headers,
+            HttpStatusCode status,
+            WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
```
Comment (d1sam) [inline src/main/java/mate/academy/onlinebookstore/exception/CustomGlobalExceptionHandler.java]: try to avoid String literals

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, but some changes should be done, please look at my comments.

File: src/main/java/mate/academy/onlinebookstore/exception/CustomGlobalExceptionHandler.java (hunk 2 above)
Comment (MinoDentori) [reply src/main/java/mate/academy/onlinebookstore/exception/CustomGlobalExceptionHandler.java]: fixed

File: src/main/java/mate/academy/onlinebookstore/dto/CreateBookRequestDto.java (hunk 1 above)
Comment (MinoDentori) [reply src/main/java/mate/academy/onlinebookstore/dto/CreateBookRequestDto.java]: fixed

File: src/main/java/mate/academy/onlinebookstore/exception/CustomGlobalExceptionHandler.java
Comment (MinoDentori) [reply src/main/java/mate/academy/onlinebookstore/exception/CustomGlobalExceptionHandler.java:1]: fixed

File: src/main/java/mate/academy/onlinebookstore/dto/CreateBookRequestDto.java
Comment (MinoDentori) [reply src/main/java/mate/academy/onlinebookstore/dto/CreateBookRequestDto.java:1]: fixed

**Review by kshuryhin — APPROVED** [review]
great job

**Pull request merged**